tests/preprocessing/test_positionfixes.py

```diff
@@ -1,116 +1,225 @@
 
-        assert len(tpls) > 0
-        assert len(tpls) == len(tpls)
+        assert (tpls.geometry.geom_almost_equals(testdata_tpls_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(tpls.drop('geom', axis=1), testdata_tpls_geolife.drop('geom', axis=1),
+                                      check_like=True)
 
-        distance_sum = 0
-        for i in range(len(tpls)):
-            distance = tpls.geom.iloc[i].distance(tpls_test.geom.iloc[i])
-            distance_sum = distance_sum + distance
-        np.testing.assert_almost_equal(distance_sum, 0.0)
-    
-    def test_generate_triplegs_dtype_consistent(self):
-        pfs_file = os.path.join('tests', 'data', 'positionfixes.csv')
-        pfs = ti.read_positionfixes_csv(pfs_file, sep=';', tz='utc', index_col='id')
-        pfs, spts = pfs.as_positionfixes.generate_staypoints(method='sliding', 
-                                                             dist_threshold=25, 
-                                                             time_threshold=5 * 60)
+    def test_generate_triplegs_case2(self, geolife_pfs_spts_short, testdata_tpls_geolife):
+        """Test tripleg generation using the 'between_staypoints' method for case 2.
+        case 2: triplegs from positionfixes (without column 'staypoint_id') and staypoints."""
+
+        pfs, spts = geolife_pfs_spts_short
+        pfs.drop('staypoint_id', axis=1, inplace=True)
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
+
+        assert (tpls.geometry.geom_almost_equals(testdata_tpls_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(tpls.drop('geom', axis=1), testdata_tpls_geolife.drop('geom', axis=1),
+                                      check_like=True)
+
+    def test_generate_triplegs_case3(self, geolife_pfs_spts_short, testdata_tpls_geolife):
+        """Test tripleg generation using the 'between_staypoints' method for case 3.
+        case 3: triplegs from positionfixes with column 'staypoint_id' but without staypoints."""
+
+        pfs, spts = geolife_pfs_spts_short
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(method='between_staypoints')
+
+        assert (tpls.geometry.geom_almost_equals(testdata_tpls_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(tpls.drop('geom', axis=1), testdata_tpls_geolife.drop('geom', axis=1),
+                                      check_like=True)
+
+    def test_tripleg_ids_in_positionfixes_case1(self, geolife_pfs_spts_short, testdata_pfs_ids_geolife):
+        """
+        checks if positionfixes are not altered and if staypoint/tripleg ids where set correctly during tripleg
+        generation using the 'between_staypoints' method with positionfixes with 'staypoint_id' and staypoints as
+        input (case 2).
+        """
+        pfs, spts = geolife_pfs_spts_short
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
```

Comment:
`tpls` is not used, use `_`.
Also Line 176,Line  191.

Reply:
👍